builtin/mainmenu/tab_settings.lua

```diff
@@ -389,6 +437,18 @@ local function handle_settings_buttons(this, fields, tabname, tabdata)
 		core.setting_set("anisotropic_filter", "true")
 		ddhandled = true
 	end
+	
+	if fields["dd_resolution"] then
+		core.setting_set("screenW",
+			video_mode_to_option_w(fields["dd_resolution"]))
+		core.setting_set("screenH",
+			video_mode_to_option_h(fields["dd_resolution"]))
+		if fields["dd_resolution"] == default_label then
```

Comment:
Addendum : the *else* can't work because everything you select in the dropdown is an action (hence the condition at line 441). Therefore the default option can't be exteriorized of that condition.